client/connections/ConnectionToCore.ts

```diff
@@ -81,18 +83,25 @@ export default abstract class ConnectionToCore {
   public async disconnect(fatalError?: Error): Promise<void> {
     if (this.isClosing) return;
     this.isClosing = true;
-    const logid = log.stats('ConnectionToCore.Disconnecting');
+    const logid = log.stats('ConnectionToCore.Disconnecting', {
+      host: this.hostOrError,
+      sessionId: null,
+    });
 
-    this.cancelPendingRequests();
+    await this.cancelPendingRequests();
     if (this.connectPromise) {
-      await this.internalSendRequestAndWait({
-        command: 'disconnect',
-        args: [fatalError],
-      });
+      await this.internalSendRequestAndWait(
+        {
+          command: 'disconnect',
+          args: [fatalError],
+        },
+        2e3,
+      );
     }
     await this.destroyConnection();
-    log.stats('RemoteConnectionToCore.Disconnected', {
+    log.stats('ConnectionToCore.Disconnected', {
```

Comment:
Why drop "Remote"? Just curious.

Reply:
This one was mis-labeled. It's inside ConnectionToCore, not RemoteConnectionToCore.